extensions/kind/src/kind-cluster-watcher.ts

```diff
@@ -0,0 +1,166 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { KubeConfig, KubernetesObject, ListPromise, V1Node, V1Pod } from '@kubernetes/client-node';
+import { CoreV1Api, makeInformer } from '@kubernetes/client-node';
+import type { Disposable } from '@podman-desktop/api';
+
+/**
+ * Utility class to manage informer lifecycle for Kind cluster readiness checks
+ */
+export class KindClusterWatcher implements Disposable {
+  private informers: Map<string, { stop?: () => void }> = new Map();
```

Comment:
Now that you don't want to get back a specific informer based on its path, but just go through tall of hem in the dispose, you may use an array here